android/app/src/main/java/com/zulipmobile/MainApplication.java

```diff
@@ -1,53 +1,64 @@
 import com.facebook.soloader.SoLoader;
+import com.learnium.RNDeviceInfo.RNDeviceInfo;
 import com.smixx.fabric.FabricPackage;
-import com.crashlytics.android.Crashlytics;
-import io.fabric.sdk.android.Fabric;
+import com.wix.reactnativenotifications.RNNotificationsPackage;
+import com.wix.reactnativenotifications.core.AppLaunchHelper;
+import com.wix.reactnativenotifications.core.AppLifecycleFacade;
+import com.wix.reactnativenotifications.core.JsIOHelper;
+import com.wix.reactnativenotifications.core.notification.INotificationsApplication;
+import com.wix.reactnativenotifications.core.notification.IPushNotification;
+import com.zmxv.RNSound.RNSoundPackage;
+import com.zulipmobile.notifications.GCMPushNotifications;
 
 import java.util.Arrays;
 import java.util.List;
 
-import com.learnium.RNDeviceInfo.RNDeviceInfo;
+public class MainApplication extends Application implements ReactApplication, INotificationsApplication {
+
+    private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
+        @Override
+        public boolean getUseDeveloperSupport() {
+            return BuildConfig.DEBUG;
+        }
 
-public class MainApplication extends Application implements ReactApplication {
+        @Override
+        protected List<ReactPackage> getPackages() {
+            return Arrays.<ReactPackage>asList(
+                    new FabricPackage(),
+                    new MainReactPackage(),
+                    new RNSoundPackage(),
+                    new RNDeviceInfo(),
+                    new ZulipNativePackage(),
+                    new RNNotificationsPackage(MainApplication.this)
+            );
+        }
+
+    };
+
+    @Override
+    public ReactNativeHost getReactNativeHost() {
+        return mReactNativeHost;
+    }
 
-  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
     @Override
-    public boolean getUseDeveloperSupport() {
-      return BuildConfig.DEBUG;
+    public void onCreate() {
+        super.onCreate();
+        SoLoader.init(this, /* native exopackage */ false);
+        //Fabric.with(this, new Crashlytics());
     }
 
     @Override
-    protected List<ReactPackage> getPackages() {
-      return Arrays.<ReactPackage>asList(
-          new FabricPackage(),
-          new MainReactPackage(),
-          new RNSoundPackage(),
-          new RNDeviceInfo(),
-          new ZulipNativePackage()
-      );
+    public IPushNotification getPushNotification(Context context, Bundle bundle, AppLifecycleFacade defaultFacade, AppLaunchHelper defaultAppLaunchHelper) {
+        return new GCMPushNotifications(context, bundle, defaultFacade, defaultAppLaunchHelper, new JsIOHelper());
     }
-  };
-
-  @Override
-  public ReactNativeHost getReactNativeHost() {
-    return mReactNativeHost;
-  }
-
-  @Override
-  public void onCreate() {
-    super.onCreate();
-    SoLoader.init(this, /* native exopackage */ false);
-    //Fabric.with(this, new Crashlytics());
-  }
 }
```

Comment:
The diff of the whole file is too random, while the actual changes seem to be few. Try not doing this.